zanata-war/src/main/java/org/zanata/servlet/MultiFileUploadServlet.java

```diff
@@ -242,29 +243,28 @@ private void respondWithFiles(HttpServletResponse response, JSONArray filesJson)
 
         private String projectSlug;
         private String versionSlug;
+        private final List<String> fileTypes;
         private String path = "";
         private String lang = "en-US";
         private String fileParams = "";
 
         private SourceDocumentUpload sourceUploader;
-        private TranslationFileService translationFileServiceImpl;
 
         public FileUploadRequestHandler(HttpServletRequest request) {
             this.request = request;
             projectSlug = request.getParameter("p");
             versionSlug = request.getParameter("v");
-
+            //TODO: alex: add types parameter in all caller of /files/upload
+            String fileTypesQuery = request.getParameter("types");
```

Comment:
- [ ] Is this something a user might type in? We should make sure it fits the expected pattern before we split it up.